File: server/router/pasKpi.go
package router

import (
	"gin-vue-admin/api/v1"
	"gin-vue-admin/middleware"
	"github.com/gin-gonic/gin"
)

func InitPasKpiRouter(Router *gin.RouterGroup) {
	PasKpiRouter := Router.Group("pasKpi").Use(middleware.JWTAuth()).Use(middleware.CasbinHandler()).Use(middleware.OperationRecord())
	{
		PasKpiRouter.POST("createPasKpi", v1.CreatePasKpi)   // 新建PasKpi
		PasKpiRouter.DELETE("deletePasKpi", v1.DeletePasKpi) // 删除PasKpi
		PasKpiRouter.DELETE("deletePasKpiByIds", v1.DeletePasKpiByIds) // 批量删除PasKpi
		PasKpiRouter.PUT("updatePasKpi", v1.UpdatePasKpi)    // 更新PasKpi
		//PasKpiRouter.PUT("updatePasKpiByIds", v1.UpdatePasKpiByIds) // 樊批量归档PasKpi
		PasKpiRouter.GET("findPasKpi", v1.FindPasKpi)        // 根据ID获取PasKpi
		PasKpiRouter.GET("getPasKpiList", v1.GetPasKpiList)  // 获取PasKpi列表
	}
}

File: web/src/api/pasKpi.js
import service from '@/utils/request'

// @Tags PasKpi
// @Summary 创建PasKpi
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.PasKpi true "创建PasKpi"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /pasKpi/createPasKpi [post]
export const createPasKpi = (data) => {
     return service({
         url: "/pasKpi/createPasKpi",
         method: 'post',
         data
     })
 }


// @Tags PasKpi
// @Summary 删除PasKpi
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.PasKpi true "删除PasKpi"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /pasKpi/deletePasKpi [delete]
 export const deletePasKpi = (data) => {
     return service({
         url: "/pasKpi/deletePasKpi",
         method: 'delete',
         data
     })
 }

// @Tags PasKpi
// @Summary 删除PasKpi
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.IdsReq true "批量删除PasKpi"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /pasKpi/deletePasKpi [delete]
 export const deletePasKpiByIds = (data) => {
     return service({
         url: "/pasKpi/deletePasKpiByIds",
         method: 'delete',
         data
     })
 }

// 樊新增批量归档
//  export const updatePasKpiByIds = (data) => {
//     return service({
//         url: "/pasKpi/updatePasKpiByIds",
//         method: 'update',
//         data
//     })
// }


// @Tags PasKpi
// @Summary 更新PasKpi
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.PasKpi true "更新PasKpi"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"更新成功"}"
// @Router /pasKpi/updatePasKpi [put]
 export const updatePasKpi = (data) => {
     return service({
         url: "/pasKpi/updatePasKpi",
         method: 'put',
         data
     })
 }


// @Tags PasKpi
// @Summary 用id查询PasKpi
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.PasKpi true "用id查询PasKpi"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"查询成功"}"
// @Router /pasKpi/findPasKpi [get]
 export const findPasKpi = (params) => {
     return service({
         url: "/pasKpi/findPasKpi",
         method: 'get',
         params
     })
 }


// @Tags PasKpi
// @Summary 分页获取PasKpi列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.PageInfo true "分页获取PasKpi列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /pasKpi/getPasKpiList [get]
 export const getPasKpiList = (params) => {
     return service({
         url: "/pasKpi/getPasKpiList",
         method: 'get',
         params
     })
 }